jmx/plugins/org.jboss.tools.jmx.ui/src/org/jboss/tools/jmx/ui/internal/wizards/DefaultConnectionWizardPage.java

```diff
@@ -275,8 +303,27 @@ protected String getNextName() {
 			return Messages.DefaultConnectionWizardPage_Default_Name;
 	}
 
+	private Job validateJob;
+	protected void validateJob() {
+		if( validateJob != null ) {
+			validateJob.cancel();
+		}
+		validateJob = new Job("Validate Wizard"){ //$NON-NLS-1$
```

Comment:
I think that it would be best to validate not on all keystroke but when user stops to type i usually use Databinding for that: WidgetProperties.text(SWT.Modify).observeDelayed(400, text)